### PR TITLE
Update ANGLE to 2024-11-29 (96952a9b1b3982ae1bf2528ed74034876e4bc967)

### PR DESCRIPTION
**Pull request by kkinnunen-apple**
#### b758eb27a4647ca9a826d5a7239ee823214ae162
<pre>
Update ANGLE to 2024-11-29 (96952a9b1b3982ae1bf2528ed74034876e4bc967)
<a href="https://bugs.webkit.org/show_bug.cgi?id=283818">https://bugs.webkit.org/show_bug.cgi?id=283818</a>
<a href="https://rdar.apple.com/problem/140689426">rdar://problem/140689426</a>

Reviewed by Dan Glastonbury

Contains upstream commits:
git log --oneline 5754fb80662a9f4baf682ce017052ba5c1b4cbaf..96952a9b1b3982ae1bf2528ed74034876e4bc967 --pretty=%h %s
96952a9b1b Roll Chromium from b640856c0ed4 to 9c3b7c9af896 (355 revisions)
a2d76f0399 Add GetTexLevelParameterTest.Levels end2end test
b31f367a9a Vulkan: Keep old VVL suppression until roll into Chromium
3f312d98b9 ES31: GetTexLevelParameter{if} validation for TEXTURE_2D_MULTISAMPLE
5d3d299d47 Expose the required GLES1.0/1.1 extensions in ANGLE
e10220f8d5 Manual roll vulkan-deps from 3c7156644de7 to 0e28d467e76d (59 revisions)
01dee1cbb0 Add implementation for GL_EXT_texture_storage_compression
8b7bb82cf8 Vulkan: Disable dynamic vertex input state on Qualcomm
0f9c146b65 Roll Chromium from c957bc3555b2 to b640856c0ed4 (1228 revisions)
6c1021ec7c Vulkan: Switch DescriptorSetLayout to use AtomicSharedPtr
4aaeffd8b9 Metal: Limit simulator texture size to 8k
2dc072ec71 Vulkan: Switch PipelineLayout from AtomicBind* to AtomicSharedPtr
87d6199796 Vulkan: Switch ShaderModule to use SharedPtr
6451a893b1 Fix minimum caps for texture gather offsets.
6f28cb15b4 Fix minimum caps for draw buffers and compute invocations.
5288ed3623 GL: Enable ANGLE_texture_multisample on OpenGL ES
b6b826b32b Add stubs for GL_EXT_EGL_image_storage_compression
8597a5fdcc Skip TextureMultisampleTest.MaskedDrawWithSampleID on S22
9c7ad319dd Revert &quot;Delay EGLQueryContext render buffer change&quot;
2e25ea1e72 Add a new mutex in CommandQueue to protect Vulkan Command Pool
c796c571e5 Disable BasicUniformUsageTest.* on Linux Intel WGPU for now
ea6e6b9fbd Capture/Replay: Fix GetTexLevelParameterivANGLE buffer size
239ef6801a Metal: Support ANGLE_texture_multisample
c56867ece5 Refine sRGB support
233d9ee5c3 Delay EGLQueryContext render buffer change
7a1da65f87 Fix GetTexLevelParameter validation
777fd9127d Roll Chromium from e5849daecf2c to c957bc3555b2 (609 revisions)
cecefe5343 Vulkan: Improve recreateSwapchain() error handling
f458f86590 Vulkan: Update AcquireNextImageUnlocked() implementation
ea86503c2a Vulkan: Remove vkAcquireNextImageKHR multi-threading support
b94d785394 Fix TextureMultisampleTest.CheckSamplePositions end2end test
bfa03a8465 Metal: Fix vec swizzles to bvecs on AMD
e3199b5f8e Add bison/flex binaries on Mac.
c02e01842f Add GL_ARM_rgba8
89f7fd8b55 CL/Vulkan: Skip crashing tests
18127908ca Roll Chromium from 1cae31feac1f to e5849daecf2c (818 revisions)
9eab301c58 Enable GL_KHR_texture_compression_astc_sliced_3d
cb31b8864d Vulkan: pruneDefaultBufferPools when there is excessive garbage
a5dec6fea2 Tests: ANGLETestBase subclasses non-zero initialize primitives
52fa677906 Refactor indexed parameter query validation
6b9b8239f4 Vulkan: fix un-initialized mCompatiblePresentModes
03380548cc CL/Vulkan: Fix nullptr case for async build task
9c20227585 Fix SAMPLE_MASK_VALUE indexed query validation
b856059e21 Tests: linux-trace logs return code with OOM note
21d747de9d Vulkan: Use vk::SharedPtr for SharedDescriptorSetCacheKey
0757254b4f Tests: Add crash callback to test Replay (e.g. linux-trace)
a3452c2482 Fix SAMPLE_MASK toggle validation
6544daecea Fix indentation in GetQueryParameterInfo
2f7feccd99 Tests: Fix EGLSurfaceTest.SwapWithoutAnyDraw test
bc09b1785b CL: Run a subset of CTS tests on linux
0c7c14eb20 Capture: add a sanity check to InitializeReplay4 args
8ba71e663e CL/Vulkan: Update few device caps based on VK caps
b18b0b0234 Skip couple non-deterministic angle_restricted_trace_gold_tests
8e9dc1a669 Validate anonymous struct names with namespace
8409b41e3b Report error on unsized interface block array
b76f0ae053 Roll Chromium from 29f00e74db5f to 1cae31feac1f (560 revisions)
10c2dc7a1b CL: Remove logic restricting CL version of passthrough backend
a58b35bcc6 CL/Vulkan: Implement image creation from buffer object
94eebab508 Tests: fix ProgramPipelineTest31 unitialized memory read
70d1ef67fe Vulkan: Ensure onFramebufferBoundary is called for offscreen
0dfe0a75f3 CL/VK: Add writeBuffer to staging/transfer routine
8e0178fb66 Vulkan: Switch SamplerBinding to Use SharedPtr
a41b798ee4 WebGPU: Ensure mDefaultBindGroup is created
8f07cdde00 Fix NegativeTextureMultisampleTest
e1fd42db9f CL/Vulkan: Add parent dependencies and image layout transitions
ec77386581 Qualifier in const for sampler2D fails validation
99d5d4d5b4 Tests: fix replay failure handling in linux/win-trace
7672101dab Roll Chromium from 185ab56e3a12 to 29f00e74db5f (604 revisions)
594a11acf8 Tests: Add Need For Speed: No Limits trace
770242db74 Translator: Remove the `gimage1D` base type
1f0ac74a7a Translator: Remove SubpassInputMS
c4533e0aae Translator: Remove the `double` base type
3515113e2b CL/Vulkan: Remove redundant state in CLImageVk
10e073a297 CL/Vulkan: Capture an event for async build task
ce53aff056 Vulkan: Add per descriptorSet LRU cache eviction
ecfa487492 Assert no GL errors at the end of image tests
3b3783bc77 Reland &quot;Possibly fix FixedQueue.ConcurrentPushPop flakiness&quot;
99aac4b931 Capture/Replay: Remove implied exts from RequiredExtensions
17904b43c7 Vulkan: Restrict EGL_ANDROID_front_buffer_auto_refresh support
f59d6617c0 Roll vulkan-deps from 867065ecbb6a to 3c7156644de7 (4 revisions)
ba74d33b77 Roll Chromium from 0236845acb4a to 185ab56e3a12 (679 revisions)
c4ec8dbb44 Vulkan: Expose EGL_ANDROID_front_buffer_auto_refresh
79c4ad91ab ssci: mark rapidjson as Security Critical: No
74f74b63df Vulkan: Add ContextVk::onFramebufferBoundary() function
087cc41138 Vulkan: Add mRenderer to ShareGroupVk class
d2e543c77b Vulkan: Consider PowerVR hardware coherent for framebuffer fetch
d8e183ab55 Skip the antutu_refinery perf test on Windows/Intel
8504b3cfaf Add third_party/rust/ to Chromium rolls
e961f7ab35 Skip KHR-GLES31.core.texture_stencil8.multisample on S22
d81d29e166 Revert &quot;Possibly fix FixedQueue.ConcurrentPushPop flakiness&quot;
42f9c200ca Comments: www.anglebug.com -&gt; anglebug.com
fd1f9e76c4 Roll vulkan-deps from 59ce475cae66 to 867065ecbb6a (6 revisions)
f0ebb3743d Roll SwiftShader from 4d3a7b64279f to 4074d9674b3e (1 revision)
8b29fe6533 Roll Chromium from 45f7f2245fc1 to 0236845acb4a (600 revisions)
987cc0de1d Vulkan: Add release utility for BufferViewHelper
dc8b2e2ba6 CL: Pass in memory properties from cl entry point
743dd7bfde CL: Add some event/memory helper functions
518e162e55 Fix validation for 3D depth/stencil textures
174694c14a Manual roll VK-GL-CTS from f674555ab03e to b27686793f88 (29 revisions)
828e2d190a Tests: Add Block Blast trace
f21cfcd6da Add a corner case for framebuffer fetch
8a2b60b2fa Add stubs for GL_EXT_texture_storage_compression
84b175546e Possibly fix FixedQueue.ConcurrentPushPop flakiness
51ff063ca8 Skip real_racing3 trace on Linux NVIDIA
843191f572 tools: add buganizer to DIR_METADATA
04222c1b36 Roll vulkan-deps from c93d7c04d2f7 to 59ce475cae66 (12 revisions)
0dd9041431 Roll Chromium from b3ded8346a3f to 45f7f2245fc1 (752 revisions)
15492c9bc4 Vulkan: A few workarounds for older ARM drivers
33dc1606ee Skip end2end failures on S22
f3404c4d24 Tests: improve Windows support in android_helper
9dd69aae81 Vulkan: Disable dynamic rendering on PowerVR
e3b3dd6845 Vulkan: Optimize color clears vs read-only depth/stencil switch
adb80cbba7 Vulkan: Optimize read-only depth/stencil switch after clear
2babbbd2bf Add egl config support GL_RGB10_A2 for Linux Headless

29855942a1 Vulkan: Add stubs for expose VK_EXT_image_compression_control
ba81145fb6 Vulkan: Emulate coherent framebuffer fetch everywhere
33c86fdedc Tests: Add SimCity BuildIt trace
Canonical link: <a href="https://commits.webkit.org/287209@main">https://commits.webkit.org/287209@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/b12746870b7c716dcb7963280a6abd74f5159f4f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/78610 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/57655 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/31992 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83271 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/29874 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/80743 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/66806 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/5936 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61570 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19493 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/81677 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51591 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/70139 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/41880 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/48937 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/25478 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28213 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70039 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/25853 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/84639 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/5975 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4108 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/69796 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6136 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/67568 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69050 "Found 7 new API test failures: /TestWebKit:WebKit2UserMessageRoundTripTest.WKURL, /TestWebKit:WebKit2UserMessageRoundTripTest.WKURLRequestRef, /TestWebKit:WebKit.HitTestResultNodeHandle, /WebKitGTK/TestInspectorServer:/webkit/WebKitWebInspectorServer/test-page-list, /TestWebKit:WebKit.WillSendSubmitEvent, /TestWebKit:WebKit.GeolocationBasicWithHighAccuracy, /TestWebKit:WebKit.LoadCanceledNoServerRedirectCallback (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17220 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13081 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/11568 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/5922 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/5909 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9344 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/7697 "Built successfully") | | |

### CODE REVIEW

File: Source/ThirdParty/ANGLE/.gitignore [hunk 1]
```diff
@@ -93,6 +93,7 @@
 /third_party/r8/lib
 /third_party/rapidjson/src
 /third_party/requests/src
+/third_party/rust
```
Comment (djg) [inline Source/ThirdParty/ANGLE/.gitignore:96]: It begins.

**Review by djg — APPROVED** [review]
:shipit:
